### PR TITLE
Update code with ProxyQueryInterface generic

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because {reason}.

Closes #{put_issue_number_here}.

## Changelog

```markdown
### Added
- Generics for ProxyQueryInterface and ProxyQuery
```

### CODE REVIEW

File: src/Builder/DatagridBuilder.php [hunk 1]
```diff
@@ -156,14 +156,15 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = []): Data
         if (!$query instanceof ProxyQueryInterface) {
             throw new \TypeError(sprintf('The admin query MUST implement %s.', ProxyQueryInterface::class));
         }
+        /** @phpstan-var ProxyQueryInterface<object> $query */
```
Comment (jordisala1991) [inline src/Builder/DatagridBuilder.php:159]: Is this one necessary? Isnt the admin->createQuery returning already with the generic applied?

File: src/Filter/ModelAutocompleteFilter.php [hunk 2]
```diff
@@ -92,7 +94,19 @@ protected function handleMultiple(ProxyQueryInterface $query, string $alias, Fil
...
+        $associationMappings[] = $this->getAssociationMapping();
+        $alias = $query->entityJoin($associationMappings);
+
+        return [$alias, ''];
+    }
+
+    /**
+     * @param ProxyQueryInterface<object> $query
+     */
+    private function handleModel(ProxyQueryInterface $query, string $alias, FilterData $data): void
```
Comment (jordisala1991) [inline src/Filter/ModelAutocompleteFilter.php:109]: isnt this change a bc break?

Comment (VincentLanglet) [reply src/Filter/ModelAutocompleteFilter.php:109]: The class is final.

File: src/Builder/DatagridBuilder.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Builder/DatagridBuilder.php:159]: It's considered as a 
```
Sonata\AdminBundle\Datagrid\ProxyQueryInterface&Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface<object>
```

And I have the error:
```
Method Sonata\DoctrineORMAdminBundle\Builder\DatagridBuilder::getBaseDatagrid() should return                                  
         Sonata\AdminBundle\Datagrid\DatagridInterface<Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface<object>> but returns  
         Sonata\AdminBundle\Datagrid\Datagrid<Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface>.  
```

**Review by jordisala1991 — APPROVED** [review]
I guess we need similar change for mongodb

**Pull request merged**